packages/core/src/stateUtils.ts

```diff
@@ -1647,22 +1647,6 @@ export function macrostep(
     microstates.push(microstate);
   }
 
-  // Handle stop event
-  if (event.type === XSTATE_STOP) {
-    nextSnapshot = cloneMachineSnapshot(
-      stopChildren(nextSnapshot, event, actorScope),
-      {
-        status: 'stopped'
-      }
-    );
-    addMicrostate(nextSnapshot, event, []);
-
-    return {
```

Comment:
The main problem was that this was returning too early, without giving the machine a chance to process the `"xstate.stop"` event